tools/fluent/basic_syntax.md

```diff
@@ -0,0 +1,131 @@
+# Introduction to Fluent syntax
+
+<!-- toc -->
+
+## Basic syntax
+
+A *message* represents the basic unit of translation in Fluent. This is the simplest form a message could take:
+
+```PROPERTIES
+welcome-msg = Hello, World!
+```
+
+`welcome-msg` is the message identifier (or message ID), and `Hello, World!` is the **value** of this message.
+
+A message can contain **external arguments**, for example:
+
+```PROPERTIES
+welcome-msg = Welcome { $user }
+```
+
+`$user` is the name of the argument, and should never be translated. The fragment included between curly braces is called a **placeable**, and can be moved within the text. For example, the string above would be translated in Italian as follows, leaving the placeable unchanged:
+
+```PROPERTIES
+welcome-msg = Benvenuto { $user }
+```
+
+Messages can also reference other messages. In this case, the placeable won’t include the `$` sign, simply the message identifier:
+
+```PROPERTIES
+menu-settings = Settings
+help-menu-settings = Click “{ menu-settings }” to save the file.
+```
+
+There is a special category of messages, called **private messages**:
+* Private messages have identifiers starting with a dash, e.g. `-brand-short-name`.
+* They can’t be called directly in the source code, but only referenced within other messages.
+
+For example:
+
+```PROPERTIES
+-brand-short-name = Firefox
+close-msg = Close { -brand-short-name }
+```
```

Comment:
Yeah, I think this calls for an explanation as to why the brand name is not a public message here. Something like:

> The reason why we're using a private message here, `-brand-short-name`, is that private message can have any number of localizer-defined attributes which can be used to encode grammatical properties of the message. Mozilla tools like `compare-locales` won't report those attributes as obsolete if the `en-US` source doesn't have them. You're free to put information about genders, animacy, the first letter being a vowel or not and any more into attributes of private messages.